sotodlib/hwp/g3thwp.py

```diff
@@ -790,21 +790,20 @@ def _calc_angle_linear(self):
             direction = self._force_quad
 
         self._encd_cnt_split = np.split(self._encd_cnt, self._ref_indexes)
-        self._angle = (self._encd_cnt_split[0] - self._ref_cnt[0]) * \
+        angle_first_revolution = (self._encd_cnt_split[0] - self._ref_cnt[0]) * \
             (2 * np.pi / self._num_edges) % (2 * np.pi)
-        self._angle = np.append(self._angle,
-                                np.concatenate(
-                                    np.array(
-                                        [((self._encd_cnt_split[i] - self._ref_cnt[i]) *
-                                          (2 * np.pi /
-                                            np.diff(self._ref_indexes)[i - 1])
-                                            % (2 * np.pi)).flatten()
-                                            for i in range(1, len(self._encd_cnt_split) - 1)], dtype=object)))
-        self._angle = np.append(self._angle,
-                                (self._encd_cnt_split[-1] - self._ref_cnt[-1]) *
-                                (2 * np.pi / self._num_edges) % (2 * np.pi))
-        self._angle = direction * self._angle % (2 * np.pi)
-
+        angle_last_revolution = (self._encd_cnt_split[-1] - self._ref_cnt[-1]) * \
+            (2 * np.pi / self._num_edges) % (2 * np.pi) + len(self._ref_cnt) * 2 * np.pi
+        self._angle = np.array([((self._encd_cnt_split[i] - self._ref_cnt[i]) * \
+                          (2 * np.pi / np.diff(self._ref_indexes)[i - 1]) \
+                              % (2 * np.pi) + i * 2 * np.pi ).flatten() \
+                                  for i in range(1, len(self._encd_cnt_split) - 1)])
+        self._angle = np.concatenate([angle_first_revolution, self._angle.flatten(), angle_last_revolution])
+        self._angle = direction * self._angle
+        if mod2pi:
+            self._angle = self._angle % (2 * np.pi)
```

Comment:
% (2*np.pi) is already applied at previous process.
Original motivation of this line is to deal in case of direction = -1.
So we should do like below if we want to implement mod2pi option:

self._angle = direction * self._angle % (2 * np.pi)
if not mod2pi: self._angle = np.unwrap(self._angle)

I think it's ok to add this option, but user can easily do this outside of this script.
What do you think?

Reply:
I added this option not to do the mod2pi.
My opinion is that "Not doing mod2pi on angle at all" and "np.unwrap(angle%2pi)" is not an equivalent operation and former better for detecting some kinds of error.

Actually, I didn't know about the `np.unwrap` so I think it's fine to delete this optio. 
I agree that this is minor point.

Reply:
Ah I agree that "not doing" and "unwrap" is not completely same for debug point of view.
OK, let's add this option.